app/components/nav.jsx

```diff
@@ -15,15 +16,53 @@ export class Nav extends Component {
     clearToken();
   }
 
+  constructor (props) {
+    super(props);
+    this.state = { loading: false };
+  }
+
+  componentWillMount () {
+    this.reset();
+  }
+
+  reset (props) {
+    const { retrieveUser, session } = props || this.props;
+
+    if (session) {
+      this.setState({ ...this.state, loading: true });
+
+      retrieveUser(session.username)
+      .then(() => this.setState({ ...this.state, loading: false }))
+      .catch(() => this.setState({ ...this.state, loading: false }));
+    }
+  }
+
   render () {
-    const { session } = this.props;
+    const { session, user } = this.props;
+    const { loading } = this.state;
+
+    if (loading) {
+      return (
+        <nav>
+          <Link to="/">Pokédex Tracker</Link>
```

Comment:
originally i had this whole thing as `null`, but changed to this to prevent as much "flickering" as possible. maybe there's a better way to avoid that with the whole better loading thing... idk